lib/def.js

```diff
@@ -173,6 +181,20 @@
           return this.base
         }
         return new infer.Arr(types || inner)
+      } else if (this.eat("{")) {
+        var types = [], names = [], computed = false
+        if (!this.eat("}")) {
+          computed = this.parseList(comp, types, names)
```

Comment:
This will allow triple dots in property names, though. And it'll disallow quoted property types, etc. Not sure if the logic for parsing object fields and function arguments is really similar enough.